applications/FSIapplication/custom_utilities/variable_redistribution_utility.cpp

```diff
@@ -59,28 +59,34 @@ void VariableRedistributionUtility::CallSpecializedConvertDistributedValuesToPoi
     const Variable<TValueType>& rDistributedVariable,
     const Variable<TValueType>& rPointVariable)
 {
-    // This function only dispatches the call to the correct specialization
-    Geometry< Node<3> >& rReferenceGeometry = rModelPart.ConditionsBegin()->GetGeometry();
-    const GeometryData::KratosGeometryFamily GeometryFamily = rReferenceGeometry.GetGeometryFamily();
-    const unsigned int PointsNumber = rReferenceGeometry.PointsNumber();
+    // Check if there is any condition in the current partition
+    int n_conds = rModelPart.GetCommunicator().LocalMesh().NumberOfConditions();
+    rModelPart.GetCommunicator().SumAll(n_conds);
 
-    if (GeometryFamily == GeometryData::Kratos_Linear && PointsNumber == 2)
-    {
-        VariableRedistributionUtility::SpecializedConvertDistributedValuesToPoint<GeometryData::Kratos_Linear,2,TValueType>(
-            rModelPart,
-            rDistributedVariable,
-            rPointVariable);
-    }
-    else if (GeometryFamily == GeometryData::Kratos_Triangle && PointsNumber == 3)
-    {
-        VariableRedistributionUtility::SpecializedConvertDistributedValuesToPoint<GeometryData::Kratos_Triangle,3,TValueType>(
-            rModelPart,
-            rDistributedVariable,
-            rPointVariable);
-    }
-    else
-    {
-        KRATOS_ERROR << "Unsupported geometry type with " << PointsNumber << " points." << std::endl;
+    // If there is conditions, this function dispatches the call to the correct specialization
+    if (n_conds != 0){
+        Geometry< Node<3> >& rReferenceGeometry = rModelPart.ConditionsBegin()->GetGeometry();
```

Comment:
This is problematic. The local model part might not have conditions, in which case this call will fail. We probably need to synchronize the values of GeometryFamily and PointsNumber to be safe.

Reply:
I think the solution would be to check both for local and global num_conds

Reply:
@philbucher This could work, but then we cannot perform collective MPI operations within the function call (because the process that has no local conditions will never enter it).

Reply:
Totally agree, I did came across with this problem during this morning... I'm currently working on it.